### PR TITLE
Improve WorldToScreen-ing points in front but off to the side

### PR DESCRIPTION
**Pull request by 0x0ade**
Would've loved to show an example, but I didn't get any pics or gifs before the patch hit, so...

This PR splits the logic statement at the end of `WorldToScreen` apart into two, allowing for the user of that method to only care if something's "in front". This is important if you want to f.e. draw a line that goes from a world point that's on-screen to another one that's still in front of the camera, but off to the side of the screen. The previous (current) "screen culling" behavior of that method makes it impossible.

I've tested this change myself for months now using a plugin that's still in-dev and a stupid reflection hack to get the matrix singleton myself, with a slightly different signature:

```cs
public void WorldToScreen(Vector3 worldPos, out Vector2 screenPos, out bool inView, out bool inFront)
```

```cs
inFront = pCoords.Z > 0;
inView = inFront &&
    screenPos.X > windowPos.X && screenPos.X < windowPos.X + width &&
    screenPos.Y > windowPos.Y && screenPos.Y < windowPos.Y + height;
```

```cs
Service.GameGuiExt.WorldToScreen(target.Position, out Vector2 to, out bool isToInView, out bool isToInFront);

if ((Service.Config.CullAggressively && !isToInView) || !isToInFront)
{
    continue;
}
```

I've decided to keep the old behavior in tact with the existing overload and adjusted the documentation to match the existing behavior accordingly, to try to avoid surprising anyone else who used that method and relied on the "screen culling" behavior. Using the overload with the additional out param and even just discarding it works fine for getting offscreen-but-infront points.

**TL;DR:** in front != in view, ~~valve~~ pls fix

### CODE REVIEW

**Comment by 0x0ade** [comment]
Force-pushed as I rebased against latest master.

**Review by goaaats — APPROVED** [review]
